### PR TITLE
enhancement: implementing design in docs

### PR DESCRIPTION
**Pull request by MelSumner**
## :pushpin: Summary

If merged, this PR implements [the design](https://www.figma.com/proto/KoYrevqh4ExBf0bMsZ5Lxi/Flight-Microsite?page-id=1%3A210&node-id=1%3A211&viewport=469%2C48%2C0.5&scaling=min-zoom) for the microsite. It should be more or less responsive, may need a few tweaks here and there.

Note: I did a lot of this work before seeing the separate, related issues opened on the repo.

## :hammer_and_wrench: Detailed Description
- [x] Navbar: setup base parts of navbar component (#165 )
- [x] Navbar Component: add HC logo 
- [x] Navbar Component: wording in navbar- words should match route names 
- [x] Icon Page: added intro cards & refactored code to look more like the mockup (#167 )
- [x] Icon Page: where is GH logo? used `git-pull-request` icon for now
- [x] Icon Page: discuss h1-h3 gap on icon page (needs to be h1 -> h2 -> h3) 
- [x] Icon Page: need to update descriptive text in boxes
- [x] Icon Page: implement toggle ( #169 ) (PR #182)
- [x] Icon Page: Hook up toggle to icon display grid (PR #199)
- [x] CSS: separated relevant css into its own file (navigation, typography)
- [x] CSS: turned on tailwind css 
- [x] CSS: `main.ds-main`: constrained content width
- [x] CSS: started removing some CSS that tailwind will replace
- [x] CSS: import our `:root` CSS variables into tailwind for class use (specifically, colors) (PR #173 )
- [x] CSS: tidy up (tracking separately: https://github.com/hashicorp/flight/issues/203) 
- [x] Design: implement tailwind layout in /design page (PR #187 )
- [x] Design: Add sidebar/aside (tracking with https://github.com/hashicorp/flight/issues/206)
- [x] Design: Added sections with section links
- [x] Design: Include Contributing section (#175 )
- [x] Design: edit images (crop to just image, put text into DOM)
- [x] Engineering: implement tailwind layout in /engineering page (https://github.com/hashicorp/flight/pull/190) 
- [x] Engineering: Make code samples copyable (https://github.com/hashicorp/flight/pull/191)
- [x] Footer Component: implement site footer
- [x] Footer component: implement "way to contact us" per updated design comp (https://github.com/hashicorp/flight/pull/201)
- [x] nitpick: rename navigation component in dummy app for consistency (tracking with https://github.com/hashicorp/flight/issues/205) 

### Notes
- used the [`@apply`](https://tailwindcss.com/docs/functions-and-directives#apply) so that our class names could remain the same
- I tried to stick to the values in `:root` and supplement with tailwind styles. 
- I'm trying to leave the base styles in the CSS on the class, but then if I need contextual styling, I will use the tailwind helper (for context-specific margin, padding, etc). I don't know how standard this approach is so we can check as a team to see if there are any objections to that approach.

### Editing Files

For folks who are new to the project, here's where to find the files that you might want to edit: 
- File to edit the [Design Page](https://github.com/hashicorp/flight/blob/main/ember-flight-icons/tests/dummy/app/templates/design.hbs)
- File to edit the [Engineering Page](https://github.com/hashicorp/flight/blob/main/ember-flight-icons/tests/dummy/app/templates/engineering.hbs)
- File to edit the [Navbar](https://github.com/hashicorp/flight/blob/main/ember-flight-icons/tests/dummy/app/components/dummy-nav.hbs) 

## :camera_flash: Screenshots

So far (9/10/21, 2:32PM CT):
image

### CODE REVIEW

File: yarn.lock
Comment (Dhaulagiri) [inline yarn.lock:1]: i don't think this file should be here ✂️

**Comment by Dhaulagiri** [comment]
👍  this looks great to me. I say we :shipit:  as long as @heatherlarsen is 👍  and then keep cranking on the other pieces

File: yarn.lock
Comment (MelSumner) [reply yarn.lock:1]: Can you help me understand how this would have gotten into the commit if it's not supposed to be there? I didn't add it explicitly. 

I'm asking because I didn't do it on purpose, which means it's likely to happen again unless I know how to stop it.

Comment (Dhaulagiri) [reply yarn.lock:1]: the only thing I can think of is if you were to inadvertently run `yarn` from the project root thinking you were actually in `ember-flight-icons`

File: ember-flight-icons/tests/dummy/app/templates/index.hbs
Comment (Dhaulagiri) [inline ember-flight-icons/tests/dummy/app/templates/index.hbs:37]: Do we know what this will eventually link to?

File: yarn.lock
Comment (MelSumner) [reply yarn.lock:1]: that makes sense, ty!

**Comment by heatherlarsen** [comment]
This is looking great! Thanks so much for getting this turned around so quickly :) 

**Answers**

- GitHub icon came from the marketing icon set (apologies, I thought it was coming from Flight). Looks like Flight Service icons aren't available in the set yet. I'm good with using the icon you chose in the meantime though. @didoo do you know when these will be available?
- Do the card title's need to be headings? Can they just be styled as such?
- Will send you an updated version with contact details in footer

**Feedback** 

- issue [nit, non-blocking]: Are you able to style the nav items to have the same padding? Right now it looks like they're all the same size and since Engineering is longer the space between them looks a little off. 
- issue [nit, non-blocking]: Can the body copy in the cards be aligned to the title in the card?
- suggestion [nit, non-blocking]: Is it possible to remove the hover from just the logo?
- suggestion [nit, non-blocking]: Add an underline to the card links on hover
- suggestion [non-blocking]: What do you think about moving the "Available Icons" above the search so the search and icons are more coupled with that heading?
- question: Where did we land on showing 24 by default and switching to 16 based on the toggle?

**Comment by MelSumner** [comment]
@heatherlarsen 

> * Do the card title's need to be headings? Can they just be styled as such?

Sure, I can do that. 

> * Will send you an updated version with contact details in footer

Thank you! 

> * issue [nit, non-blocking]: Are you able to style the nav items to have the same padding? Right now it looks like they're all the same size and since Engineering is longer the space between them looks a little off.

I agree, I'll work on it. 

> * issue [nit, non-blocking]: Can the body copy in the cards be aligned to the title in the card?

it's a little bit of a pain but I will do it

> * suggestion [nit, non-blocking]: Is it possible to remove the hover from just the logo?

yes, can do

> * suggestion [nit, non-blocking]: Add an underline to the card links on hover

yes, can do that too

> * suggestion [non-blocking]: What do you think about moving the "Available Icons" above the search so the search and icons are more coupled with that heading?

Sounds good! I'll do it. 

> * question: Where did we land on showing 24 by default and switching to 16 based on the toggle?

Shouldn't we show 16 by default since it is the default? Other than that, I just haven't implemented the toggle yet.

File: ember-flight-icons/tests/dummy/app/templates/index.hbs
Comment (MelSumner) [reply ember-flight-icons/tests/dummy/app/templates/index.hbs:36]: yes, I will add it 👍

**Comment by heatherlarsen** [comment]
> Shouldn't we show 16 by default since it is the default? Other than that, I just haven't implemented the toggle yet.

heh. that is a fair point and an oversight on my part. 16 by default would be ideal. i can fix it in the mocks just so they're a correct representation. 

I have some really nitpicky spacing things, if you have bandwidth: 

- 3.25rem margin-bottom for the card container
- 3rem margin-bottom for the form group
- 0.75rem space between the H2 and form group

Otherwise, everything else looks great!

File: ember-flight-icons/tests/dummy/app/templates/engineering.hbs
Comment (didoo) [inline ember-flight-icons/tests/dummy/app/templates/engineering.hbs:98]: reminder: later when the new packages will be released, we'll have to update this section with a more detailed explanation of how to use them (or alternatively, point to the README in the repo if we want to favour the usage as Ember Addon)

File: ember-flight-icons/tests/dummy/app/styles/app.css
Comment (didoo) [inline ember-flight-icons/tests/dummy/app/styles/app.css:3]:
```suggestion
```

do we still need this comment?

**Review by didoo — APPROVED** [review]
I have added a few comments. 

Regarding the need of removing extra CSS selectors, it's not a blocker or an issue per se, likely there are good reasons that I am missing, but my "conscience" didn't allow me to not notice it :) .

Overall, amazing improvement. Good job @MelSumner @heatherlarsen! 👌💯

File: ember-flight-icons/tests/dummy/app/styles/app.css
Comment (MelSumner) [reply ember-flight-icons/tests/dummy/app/styles/app.css:13]: @didoo I agree in general but I would like us to review all the variables in :root (I didn't make any of these up myself, I just used the ones that already existed)

**Comment by MelSumner** [comment]
@heatherlarsen @didoo @dizzyup Here's what it looks like if I take out the custom font: (left side is without, right side is how it is currently): 
image
Are we okay with that?

File: ember-flight-icons/tests/dummy/app/styles/nav.css
Comment (MelSumner) [reply ember-flight-icons/tests/dummy/app/styles/nav.css:49]: I was intentionally verbose in the CSS for the start; I figured that once we have decided on an approach in general we could figure out how to be more terse. IMO it's easier to take out extra class names then put extra class names in, and I wanted it to be easier for whoever works on these files next (even if it's still just me) :thumbsup:

**Comment by heatherlarsen** [comment]
@MelSumner I don't love it, especially the titles in the hero cards. They're way too prominent without the custom font, so maybe we drop them down a scale size lower, if we have to keep this font. 

What's your end goal here though? To get this closed out and fix later? Or to leave as is long term?

File: ember-flight-icons/tests/dummy/app/styles/app.css
Comment (MelSumner) [reply ember-flight-icons/tests/dummy/app/styles/app.css:171]: Going to track this here: https://github.com/hashicorp/flight/issues/202

File: ember-flight-icons/tests/dummy/app/styles/app.css
Comment (MelSumner) [reply ember-flight-icons/tests/dummy/app/styles/app.css:154]: Tracking here: https://github.com/hashicorp/flight/issues/202

File: ember-flight-icons/tests/dummy/app/styles/app.css
Comment (MelSumner) [reply ember-flight-icons/tests/dummy/app/styles/app.css:185]: Tracking here: https://github.com/hashicorp/flight/issues/202

File: ember-flight-icons/tests/dummy/app/styles/nav.css
Comment (MelSumner) [reply ember-flight-icons/tests/dummy/app/styles/nav.css:49]: Tracking here: https://github.com/hashicorp/flight/issues/203

File: ember-flight-icons/tests/dummy/app/styles/nav.css
Comment (MelSumner) [reply ember-flight-icons/tests/dummy/app/styles/nav.css:1]: Tracking here: https://github.com/hashicorp/flight/issues/203

File: ember-flight-icons/tests/dummy/app/styles/footer.css
Comment (MelSumner) [reply ember-flight-icons/tests/dummy/app/styles/footer.css:1]: Tracking here: https://github.com/hashicorp/flight/issues/203

File: ember-flight-icons/tests/dummy/app/styles/app.css
Comment (MelSumner) [reply ember-flight-icons/tests/dummy/app/styles/app.css:131]: Tracking here: https://github.com/hashicorp/flight/issues/203

File: ember-flight-icons/tests/dummy/app/templates/engineering.hbs
Comment (MelSumner) [reply ember-flight-icons/tests/dummy/app/templates/engineering.hbs:98]: Tracking the issue here: https://github.com/hashicorp/flight/issues/204

**Comment by MelSumner** [comment]
> @MelSumner I don't love it, especially the titles in the hero cards. They're way too prominent without the custom font, so maybe we drop them down a scale size lower, if we have to keep this font.
> 
> What's your end goal here though? To get this closed out and fix later? Or to leave as is long term?

I don't love it either, which is why I asked before shipping it. I would like to reduce the [FOUT](https://css-tricks.com/fighting-foit-and-fout-together/) on the docs site. It was mentioned in team sync this morning that maybe we should remove the custom fonts completely. I think I will look for other options and leave it for now.

File: ember-flight-icons/tests/dummy/app/templates/design.hbs
Comment (MelSumner) [reply ember-flight-icons/tests/dummy/app/templates/design.hbs:11]: Is there anything I can do now to help ship these changes? @didoo @heatherlarsen

File: ember-flight-icons/tests/dummy/app/components/ds-footer.hbs
Comment (Dhaulagiri) [inline ember-flight-icons/tests/dummy/app/components/ds-footer.hbs:14]: I think it'd be better not to publicize this specific email address externally as we don't necessarily want inbound email from outside the company

Comment (MelSumner) [reply ember-flight-icons/tests/dummy/app/components/ds-footer.hbs:14]: Should we also not link to internal slack then? 

The point of adding contact info in the footer is that it's an accessibility conformance criteria, to provide a way for users to get in contact with us in some way.

**Review by heatherlarsen — APPROVED** [review]
chatted with @MelSumner over Slack. lgtm, will iterate after this is merged up.

**Pull request merged**
